packages/arb-token-bridge-ui/src/util/TokenUtils.ts

```diff
@@ -60,86 +89,55 @@ export type FetchErc20DataProps = {
+
+  if (cachedErc20Data) {
+    return cachedErc20Data
+  }
 
   // todo: fall back if there is no multicall?
+  const multiCaller = await MultiCaller.fromProvider(provider)
   const [tokenData] = await multiCaller.getTokenData([address], {
     name: true,
     symbol: true,
     decimals: true
   })
 
-  return {
+  const erc20Data: Erc20Data = {
     name: tokenData?.name ?? getDefaultTokenName(address),
     symbol: tokenData?.symbol ?? getDefaultTokenSymbol(address),
-    decimals: tokenData?.decimals ?? defaultErc20Decimals
-  }
-}
-
-/**
- * Retrieves static data (name, decimals, symbol, address) about an ERC-20 token using its L1 address
- * @param erc20L1Address,
- * @param l1Provider
- */
-export async function getL1TokenData({
-  account,
-  erc20L1Address,
-  l1Provider,
-  l2Provider,
-  throwOnInvalidERC20 = true
-}: {
-  account: string
-  erc20L1Address: string
-  l1Provider: Provider
-  l2Provider: Provider
-  throwOnInvalidERC20?: boolean
-}): Promise<L1TokenData> {
-  // checking the cache for tokens results
-  // if successfully found in the cache, return the token data
-  const l1TokenDataCache = getTokenDataCache()
-  const cachedTokenData = l1TokenDataCache[erc20L1Address]
-  if (cachedTokenData) return cachedTokenData
-
-  // else, call on-chain method to retrieve token data
-  const contract = ERC20__factory.connect(erc20L1Address, l1Provider)
-
-  const l1GatewayAddress = await fetchErc20L1GatewayAddress({
-    erc20L1Address,
-    l1Provider,
-    l2Provider
-  })
-
-  const multiCaller = await MultiCaller.fromProvider(l1Provider)
-  const [tokenData] = await multiCaller.getTokenData([erc20L1Address], {
-    decimals: true,
-    name: true,
-    symbol: true,
-    allowance: { owner: account, spender: l1GatewayAddress } // getting allowance will help us know if it's a valid ERC20 token
-  })
-
-  if (tokenData && typeof tokenData.allowance === 'undefined') {
-    if (throwOnInvalidERC20)
-      throw new Error(
-        `getL1TokenData: No allowance method available for ${erc20L1Address}`
-      )
-  }
-
-  const finalTokenData = {
-    name: tokenData?.name ?? getDefaultTokenName(erc20L1Address),
-    symbol: tokenData?.symbol ?? getDefaultTokenSymbol(erc20L1Address),
     decimals: tokenData?.decimals ?? defaultErc20Decimals,
-    address: contract.address
+    address
   }
 
-  // store the newly fetched final-token-data in cache
   try {
-    setTokenDataCache(erc20L1Address, finalTokenData)
+    setErc20DataCache({ chainId, address, erc20Data })
   } catch (e) {
+    console.warn(`Failed to store ERC-20 data to cache.`)
     console.warn(e)
   }
 
-  return finalTokenData
+  return erc20Data
+}
+
+export async function isValidErc20({
+  address,
+  provider
+}: FetchErc20DataProps): Promise<boolean> {
+  const erc20 = ERC20__factory.connect(address, provider)
+
+  try {
+    await Promise.all([
+      erc20.balanceOf('0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045'),
```

Comment:
I sorta liked the explanation we removed for using this address. We should add it here.

`
// we don't care about allowance in this call, so we're just using vitalik.eth

    // didn't want to use address zero in case contracts have checks for it
   `